File: front/src/app/models/order-product.ts
export class OrderProduct {
  productName: string;
  quantity: number;
  price: number;
  total: number;

  constructor(productName: string, quantity: number, price: number, total: number) {
    this.productName = productName;
    this.quantity = quantity;
    this.price = price;
    this.total = total;
  }

}

File: checkout/src/main/java/com/vkuzmenko/pizza/checkout/domain/OrderProduct.java
package com.vkuzmenko.pizza.checkout.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "order_product")
public class OrderProduct {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    private String productName;
    private int quantity;
    private int price;
    private int total;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id", nullable = false)
    @JsonIgnore
    private Order order;
}
